src/contexts/SetComponents/SetComponentsProvider.tsx

```diff
@@ -0,0 +1,146 @@
+          return await convertPositionToSetComponent(position, tokenList, mviComponentPrices[position.component.toLowerCase()]?.usd, mviPrice)
+        })
+        Promise.all(mviPositions).then(sortPositionsByPercentOfSet).then(setMviComponents)
+
+        const bedComponentPrices = await getPositionPrices(bed)
+        const bedPositions = bed.positions.map(async position => {
+          return await convertPositionToSetComponent(position, tokenList, bedComponentPrices[position.component.toLowerCase()]?.usd, bedPrice)
+        })
+        Promise.all(bedPositions).then(sortPositionsByPercentOfSet).then(setBedComponents)
+
+        const eth2xfliComponentPrices = await getPositionPrices(eth2xfli)
+        const eth2xfliPositions = eth2xfli.positions.map(async position => {
+          return await convertPositionToSetComponent(position, tokenList, eth2xfliComponentPrices[position.component.toLowerCase()]?.usd, eth2xfliPrice)
+        })
+        Promise.all(eth2xfliPositions).then(sortPositionsByPercentOfSet).then(setEth2xfliComponents)
+
+        const btc2xfliComponentPrices = await getPositionPrices(btc2xfli)
+        const btc2xfliPositions = btc2xfli.positions.map(async position => {
+          return await convertPositionToSetComponent(position, tokenList, btc2xfliComponentPrices[position.component.toLowerCase()]?.usd, btc2xfliPrice)
+        })
+        Promise.all(btc2xfliPositions).then(sortPositionsByPercentOfSet).then(setBtc2xfliComponents)
+
+        const dataComponentPrices = await getPositionPrices(data)
+        const dataPositions = data.positions.map(async position => {
+          return await convertPositionToSetComponent(position, tokenList, dataComponentPrices[position.component.toLowerCase()]?.usd, dataPrice)
+        })
+        Promise.all(dataPositions).then(sortPositionsByPercentOfSet).then(setDataComponents)
+      })
+    }
+  }, [ethereum, tokenList, dpiPrice, mviPrice, bedPrice, eth2xfliPrice, btc2xfliPrice, dataPrice])
+
+  return (
+    <SetComponentsContext.Provider
+      value={{
+        dpiComponents: dpiComponents,
+        mviComponents: mviComponents,
+        bedComponents: bedComponents,
+        eth2xfliComponents: eth2xfliComponents,
+        btc2xfliComponents: btc2xfliComponents,
+        dataComponents: dataComponents,
+      }}
+    >
+      {children}
+    </SetComponentsContext.Provider>
+  )
+}
+
+async function convertPositionToSetComponent(position: Position, tokenList: Token[], componentPriceUsd: number, setPriceUsd: number): Promise<SetComponent> {
+  const token = getTokenForPosition(tokenList, position)
+  const quantity = new BigNumber(position.unit.toString()).div(new BigNumber(10).pow(getDecimalPlaces(token.symbol)));
+  const totalPriceUsd = quantity.multipliedBy(componentPriceUsd);
+  const percentOfSet = totalPriceUsd.dividedBy(setPriceUsd).multipliedBy(100)
+
+  if (token.symbol.toLowerCase() === "ceth" || token.symbol.toLowerCase() === 'usdc') {
```

Comment:
The way you can tell if a component is a debt position is by looking at the `positionState` field.
Here's an excerpt from ETH2x-FLI
![image](https://user-images.githubusercontent.com/7955024/141834665-c7d199b2-cdc9-431f-8a6e-19bfbc6c5fe2.png)
If `positionState === 0` it's a normal component
If `positionState === 1` it's a held in an external contract (likely a debt position)

Reply:
Thanks! These console.logs are from debugging. I'll remove them!